test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/partner/PartnerTypeControllerSpec.scala

```diff
@@ -107,76 +108,101 @@ class PartnerTypeControllerSpec extends ControllerSpec {
+               Seq(nominatedPartner(PartnerTypeEnum.OVERSEAS_COMPANY_UK_BRANCH))
+             )
+            ),
+            (CHARITABLE_INCORPORATED_ORGANISATION, scottishPartnershipDetails)
+          )
+        ) { partnershipDetails =>
+          s"a ${partnershipDetails._1} type was selected" in {
+            val registration = aRegistration(withPartnershipDetails(Some(partnershipDetails._2)))
+
+            authorizedUser()
+            mockRegistrationFind(registration)
+            mockRegistrationUpdate()
+
+            mockCreateSoleTraderPartnershipGrsJourneyCreation("http://test/redirect/soletrader")
+            mockCreateUkCompanyPartnershipGrsJourneyCreation("http://test/redirect/ukCompany")
+            mockCreatePartnershipGrsJourneyCreation("http://test/redirect/partnership")
+
+            val correctForm =
+              Seq("answer" -> partnershipDetails._1.toString, saveAndContinueFormAction)
+            val result = controller.submitNewPartner()(postJsonRequestEncoded(correctForm: _*))
+            partnershipDetails._1 match {
+              case SOLE_TRADER =>
+                redirectLocation(result) mustBe Some("http://test/redirect/soletrader")
+              case UK_COMPANY | OVERSEAS_COMPANY_UK_BRANCH =>
+                redirectLocation(result) mustBe Some("http://test/redirect/ukCompany")
+              case LIMITED_LIABILITY_PARTNERSHIP =>
+                redirectLocation(result) mustBe Some("http://test/redirect/partnership")
+              case _ =>
+                redirectLocation(result) mustBe Some(
+                  orgRoutes.OrganisationTypeNotSupportedController.onPageLoad().url
+                )
+            }
           }
+        }
+      }
+    }
 
-          val correctForm =
-            Seq("answer" -> partnershipDetails._1.toString, saveAndContinueFormAction)
-          val result = controller.submitNewPartner()(postJsonRequestEncoded(correctForm: _*))
-          partnershipDetails._1 match {
-            case SOLE_TRADER =>
-              redirectLocation(result) mustBe Some("http://test/redirect/soletrader")
-            case UK_COMPANY | OVERSEAS_COMPANY_UK_BRANCH =>
-              redirectLocation(result) mustBe Some("http://test/redirect/ukCompany")
-            case LIMITED_LIABILITY_PARTNERSHIP | SCOTTISH_PARTNERSHIP |
-                SCOTTISH_LIMITED_PARTNERSHIP =>
-              redirectLocation(result) mustBe Some("http://test/redirect/partnership")
-            case _ =>
-              redirectLocation(result) mustBe Some(
-                orgRoutes.OrganisationTypeNotSupportedController.onPageLoad().url
-              )
+    "redirect to capture partner name for new partner" when {
+      "user selected a type which has no GRS provided name" when {
+        forAll(
+          Seq(
+            (SCOTTISH_LIMITED_PARTNERSHIP,
+             scottishPartnershipDetails.copy(partners =
+               Seq(nominatedPartner(SCOTTISH_LIMITED_PARTNERSHIP))
+             )
+            ),
```

Comment:
Not sure about this one - I don't think we should be collecting a name for Scottish Limited Partnerships.